app/models/package_manager/maven.rb

```diff
@@ -210,11 +206,24 @@ def self.latest_version(versions, name)
       end
     end
 
+    def self.parse_names(page)
+      # parse the names from maven-repository.com/artifact/latest pages
+      names = page.css('tr')[1..-1]&.map do |tr|
+        tr.css('td')[0..1]&.map(&:text).join(':')
+      end || []
+
+      names.select(&MavenUrl.method(:legal_name?)).uniq
+    end
+
     class MavenUrl
       def self.from_name(name)
         new(*name.split(':', 2))
       end
 
+      def self.legal_name?(name)
+        name.split(':').size == 2
```

Comment:
Looks like `name` might be nil, so this will bomb - perhaps `compact` the `names` array?